Add-docker

Vexvl opened the pull request:
BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

Vexvl committed:
feat : BaseClient.java
       BookingClient.java
       BookingController.java
       BookingState.java
       BookItemRequestDto.java
       CommentClient.java
       CommentController.java
       CommentDto.java
       docker-compose.yml
       Dockerfile
       Dockerfile
       EmailDuplicateException.java
       ExceptionsHandler.java
       ItemClient.java
       ItemController.java
       ItemDto.java
       ItemRequestClient.java
       ItemRequestController.java
       ItemRequestDto.java
       ItemUnavailableException.java
       NoBookingException.java
       OwnerBookingException.java
       pom.xml
       pom.xml
       ShareItGateway.java
       UserClient.java
       UserController.java
       UserDto.java
       WrongDateBookingException.java

sergeymamonov commented on gateway/src/main/java/ru/practicum/shareit/handler/ExceptionsHandler.java:
```diff
@@ -0,0 +1,60 @@
+
+    @ExceptionHandler(EmailDuplicateException.class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    public Map<String, String> handleEmailDuplicateException(EmailDuplicateException e) {
```
А как вы проверяете, что уже используется этот email, без запроса к БД? И остальные исключения, которые здесь обрабатываются тоже проверьте, могу ли они возникнуть в сервисе gateway.

sergeymamonov commented on gateway/.gitignore:
```diff
@@ -0,0 +1,38 @@
+target/
```
Файл гитигнор должен быть только один и лежать он должен в корне проекта. Один репозиторий, один файл гитигнор.

sergeymamonov commented on pom.xml:
```diff
@@ -12,10 +12,15 @@
 	<groupId>ru.practicum</groupId>
 	<artifactId>shareit</artifactId>
 	<version>0.0.1-SNAPSHOT</version>
+    <packaging>pom</packaging>
```
Указывать зависимости в родительском помнике не очень хорошая практика. Лучше указывать все зависимости в дочерних помниках. Пример, нужна зависимость для каждого из подмодулей, но в одном нужна версия 1, а во втором версия 2. Если указывать зависимость в дочерних помниках, то легко указать в одном одну версию, а во втором другую версию.

sergeymamonov commented on server/.gitignore:
```diff
@@ -0,0 +1,38 @@
+target/
```
Лишний файл.

Vexvl replied:
Спасибо за замечания, всё исправил

sergeymamonov replied:
Не увидел изменений в BookingServiceImpl строка 49, проверка bookingDto.getStart().isBefore(bookingDto.getEnd() осталась.

sergeymamonov reviewed (changes requested):
BookingServiceImpl строка 49 эту проверку можно сделать в gateway?

Где используются исключения из пакета gateway/src/main/java/ru/practicum/shareit/item/exception

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
Merge remote-tracking branch 'origin/add-docker' into add-docker

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix : ExceptionsHandler
pom.xml
pom.xml
pom.xml
BookingController.java
ItemClient.java
ItemRequestController.java
UserController.java

Vexvl committed:
fix

Vexvl committed:
fix .gitignore
    application.properties
    BaseClient.java
    BookingClient.java
    BookingController.java
    BookingState.java
    BookItemRequestDto.java
    CommentClient.java
    docker-compose.yml
    Dockerfile
    ItemClient.java
    ItemRequestClient.java
    lombok.config
    pom.xml
    pom.xml
    pom.xml
    ShareItGateway.java
    UserClient.java

Vexvl committed:
fix .gitignore
    application.properties
    BaseClient.java
    BookingClient.java
    BookingController.java
    BookingState.java
    BookItemRequestDto.java
    CommentClient.java
    docker-compose.yml
    Dockerfile
    ItemClient.java
    ItemRequestClient.java
    lombok.config
    pom.xml
    pom.xml
    pom.xml
    ShareItGateway.java
    UserClient.java

Vexvl committed:
fix .gitignore
    application.properties
    BaseClient.java
    BookingClient.java
    BookingController.java
    BookingState.java
    BookItemRequestDto.java
    CommentClient.java
    docker-compose.yml
    Dockerfile
    ItemClient.java
    ItemRequestClient.java
    lombok.config
    pom.xml
    pom.xml
    pom.xml
    ShareItGateway.java
    UserClient.java

Vexvl committed:
fix .gitignore
    application.properties
    BaseClient.java
    BookingClient.java
    BookingController.java
    BookingState.java
    BookItemRequestDto.java
    CommentClient.java
    docker-compose.yml
    Dockerfile
    ItemClient.java
    ItemRequestClient.java
    lombok.config
    pom.xml
    pom.xml
    pom.xml
    ShareItGateway.java
    UserClient.java

Vexvl committed:
fix .gitignore
    application.properties
    BaseClient.java
    BookingClient.java
    BookingController.java
    BookingState.java
    BookItemRequestDto.java
    CommentClient.java
    docker-compose.yml
    Dockerfile
    ItemClient.java
    ItemRequestClient.java
    lombok.config
    pom.xml
    pom.xml
    pom.xml
    ShareItGateway.java
    UserClient.java

Vexvl committed:
fix .gitignore
    application.properties
    BaseClient.java
    BookingClient.java
    BookingController.java
    BookingState.java
    BookItemRequestDto.java
    CommentClient.java
    docker-compose.yml
    Dockerfile
    ItemClient.java
    ItemRequestClient.java
    lombok.config
    pom.xml
    pom.xml
    pom.xml
    ShareItGateway.java
    UserClient.java

Vexvl committed:
fix: ExceptionsHandler.java

Vexvl commented:
Извините, не до конца исправил все тесты на гитхабе

sergeymamonov commented on gateway/src/main/java/ru/practicum/shareit/handler/ExceptionsHandler.java:
```diff
@@ -0,0 +1,25 @@
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<String> handleUnhandledException(Exception e) {
+        return new ResponseEntity<>("An unexpected error occurred", HttpStatus.BAD_REQUEST);
```
Может быть планировалось INTERNAL_SERVER_ERROR возвращать?
